examples/columnar.rs

```diff
@@ -0,0 +1,453 @@
+        }
+    }
+
+    impl<C: AsBytes + Clear + Len + Clone + Default + 'static> LengthPreservingContainerBuilder for ColumnBuilder<C>
+    where
+        for<'a> C::Borrowed<'a> : Len + Index,
+    { }
+}
+
+mod merger {
+
+    use columnar::{Clear, Len, Index, Push, bytes::AsBytes};
+    use timely::{Container, container::PushInto};
+    use timely::progress::{Antichain, frontier::AntichainRef};
+    use differential_dataflow::trace::implementations::merge_batcher::container::MergerChunk;
+    use differential_dataflow::difference::{IsZero, Semigroup};
+    use crate::Column;
+
+    pub trait AsSpecificBytes : AsBytes 
+    where 
+        for<'a> Self::Borrowed<'a> : Len + Index
+    {
+        type Owned;
+        fn into_owned<'a>(item: &<Self::Borrowed<'a> as Index>::Ref) -> Self::Owned;
+        fn clone_onto<'a>(item: &<Self::Borrowed<'a> as Index>::Ref, owned: &mut Self::Owned);
+    }
+
+    impl<CD, CT, CR> MergerChunk for Column<(CD, CT, CR)> 
+    where 
+        CD: Default + Clone + Len + Clear + AsBytes + for<'a> Push<<CD::Borrowed<'a> as Index>::Ref> + 'static,
+        for<'a> CD::Borrowed<'a> : Len + Index,
+        CT: Default + Clone + Len + Clear + AsSpecificBytes + for<'a> Push<<CT::Borrowed<'a> as Index>::Ref> + 'static,
+        <CT as AsSpecificBytes>::Owned : timely::PartialOrder + for<'a> timely::PartialOrder<<CT::Borrowed<'a> as Index>::Ref>,
+        for<'a> CT::Borrowed<'a> : Len + Index,
+        for<'a> <CT::Borrowed<'a> as Index>::Ref : timely::PartialOrder<<CT as AsSpecificBytes>::Owned>,
+        CR: Default + Clone + Len + Clear + AsSpecificBytes + for<'a> Push<<CR::Borrowed<'a> as Index>::Ref> + 'static,
+        <CR as AsSpecificBytes>::Owned : Default + IsZero + for<'a> Semigroup<<CR::Borrowed<'a> as Index>::Ref>,
+        for<'a> CR::Borrowed<'a> : Len + Index,
+        CR: for<'a> Push<&'a <CR as AsSpecificBytes>::Owned>,
+    {
+        type TimeOwned = <CT as AsSpecificBytes>::Owned;
+        type DiffOwned = <CR as AsSpecificBytes>::Owned;
+
+        fn time_kept(item: &Self::Item<'_>, upper: &AntichainRef<Self::TimeOwned>, frontier: &mut Antichain<Self::TimeOwned>) -> bool {
+            let (_, time, _) = item;
+            if upper.less_equal(time) {
+                frontier.insert_with(time, |time| <CT as AsSpecificBytes>::into_owned(time));
+                true
+            } 
+            else { false }
+        }
+
+        fn push_and_add<'a>(&mut self, item1: Self::Item<'a>, item2: Self::Item<'a>, stash: &mut Self::DiffOwned) {
+            let (data, time, diff1) = item1;
+            let (___d, ___t, diff2) = item2;
+            <CR as AsSpecificBytes>::clone_onto(&diff1, stash);
+            stash.plus_equals(&diff2);
+            self.push_into((data, time, &*stash));
+        }
+
+        fn account(&self) -> (usize, usize, usize, usize) {
+            let (size, capacity, allocations) = (0, 0, 0);
+            (self.len(), size, capacity, allocations)
+        }
```

Comment:
Should implement this at some point.

Reply:
Ack, though I think it is non-trivial without mapping the same four-tuple based stuff back to columnar. If I'm reading this correctly, the `Vec` implementation currently just says `(self.len(), 0, 0, 0)` so probably not a blocker for a demo `MergerChunk` implementation at the moment?